docker/checkout.bash

```diff
@@ -37,8 +37,20 @@ done
 
 if [[ "$BRANCH" = "foxy/develop" ]]; then
       sudo git clone https://github.com/usdot-fhwa-stol/carma-msgs.git ${dir}/src/CARMAMsgs --branch $BRANCH
-      sudo git clone https://github.com/novatel/novatel_oem7_driver.git --branch ros2-dev 3055e22
+      sudo git clone https://github.com/novatel/novatel_oem7_driver.git --branch ros2-dev 3055e220bb9715b59c3ef53ab0aba05a495d9d5c
```

Comment:
Why have we duplicated these external checkouts in the if? Since they are identical in both parts why no just move them out.

Reply:
Weird that this is showing up on the pr, there is only one checkout happening for this repo